### DISCUSSION TITLE
EXIT CODE: 11

### DISCUSSION THREAD

**Discussion by DomenicoCFD** [idaholab/moose#18588] [Q&A General] [answered]
Hi All,

I am trying to resolve a 3D numerical problem on a cluster which does not differ from other for which I already got a converged solution in MOOSE. For some reasons I got the following error message:

Does anyone have an idea of what this could be related to?

I look forward to hearing from you.

Regards,

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 9079 RUNNING AT node-vvvvv5
=   EXIT CODE: 11
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================

**Comment by GiudGiud** [comment]
Hello

This could be for a variety of reasons, actual bug / running out of memory / MPI issues. 
Could you please compile in debug mode (`METHOD=dbg make -j XX`) and re-run this a debugger?
A shorthand for it:
`mpirun -n XXX gdb --ex run --ex bt --args ./your_app-dbg -i your_input_file.i`

Guillaume

**Comment by DomenicoCFD** [comment]
Hi Gulliaume,

Many thanks for your email. As I am relatively new to MOOSE, could you
please give me a bit more directions with regards to what you mean with
your command line? In fact, referring to it (see below):

mpirun -n XXX gdb --ex run --ex bt ./your_app-dbg -i your_input_file.i


   - What does XXX stand for?
   - How should I get the ./your_app-dbg? I tried to follow the
   instructions reported in the following link:

https://mooseframework.inl.gov/getting_started/examples_and_tutorials/examples/ex21_debugging.html

Therefore I initially tried to execute the following command in the main
application folder:

METHOD=dbg make -j8

but then I got the following error message:

Makefile:19: /gpfs/cfms/home/username/projects/moose/framework/build.mk: No
such file or directory
Makefile:20: /gpfs/cfms/home/username/projects/moose/framework/moose.mk: No
such file or directory
Makefile:48: /gpfs/cfms/home/username/projects/moose/modules/modules.mk: No
such file or directory
Makefile:56: /gpfs/cfms/home/username/projects/moose/framework/app.mk: No
such file or directory
make: *** No rule to make target
`/gpfs/cfms/home/username/projects/moose/framework/app.mk'.  Stop

Hence, could you please clarify more about the details on the command line
you shared?

I look forward to hearing from you.

Yours Sincerely,

Domenico di Cugno, Ph. D.











On Thu, 12 Aug 2021 at 16:36, Guillaume Giudicelli ***@***.***>
wrote:

> Hello
>
> This could be for a variety of reasons, actual bug / running out of memory
> / MPI issues.
> Could you please compile in debug mode and re-run this a debugger?
> A shorthand for it:
> mpirun -n XXX gdb --ex run --ex bt ./your_app-dbg -i your_input_file.i
>
> Guillaume
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/18588#discussioncomment-1164310>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVFIBH67FRRUJXD6JMQHHKLT4PTBNANCNFSM5CA2NXMA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
*Domenico di Cugno* // Systems Engineer

*e: ***@***.***   *t*: 0117 906 1100   *w: *
www.cfms.org.uk
CFMS Services Ltd // Bristol & Bath Science Park // Dirac Crescent // Emersons
Green // Bristol // BS16 7FR

<https://cfms.org.uk/media/384108/email-footer-darren-swift.png?width=500&height=115&mode=crop>
<https://cfms.org.uk/media/384108/email-footer-darren-swift.png?width=500&height=115&mode=crop>
<https://cfms.org.uk/industry-themes/smart-infrastructure/?utm_source=email&utm_medium=gmail-footer&utm_campaign=infrastructure&utm_content=landing-page.>
<https://cfms.org.uk/industry-themes/smart-infrastructure/?utm_source=email&utm_medium=gmail-footer&utm_campaign=infrastructure&utm_content=landing-page.>
<https://cfms.org.uk/news-events-opinions/opinions/2021/may/make-the-most-of-automation-with-open-networking/?utm_source=email&utm_medium=gmail-footer&utm_campaign=opinions&utm_content=open-networking.>
<https://cfms.org.uk/news-events-opinions/opinions/2021/may/make-the-most-of-automation-with-open-networking/?utm_source=email&utm_medium=gmail-footer&utm_campaign=opinions&utm_content=open-networking.>
<https://cfms.org.uk/news-events-opinions/opinions/2021/may/make-the-most-of-automation-with-open-networking/?utm_source=email&utm_medium=gmail-footer&utm_campaign=opinions&utm_content=open-networking.>

**Reply by GiudGiud** [reply]
XX is the number of processes here. 

note the --args I added to the command btw

Are you getting the same errors with the regular `make` command ?
Which directory are you running this from? Are you using the conda installation, if so did you remember to activate the conda environment?

Have you used gdb before? The page you linked is the right one to help you.

**Comment by DomenicoCFD** [comment, marked as answer]
Hi,

Many thanks for your email and your corrections/amendments on your previous
instructions.

I assumed I should be considering the same number of processes as that I
want to make the regular run. And I also guess that XX appearing into
METHOD ... should match the XXX that you specified into the command line
mpirun ....

Are you getting the same errors with the regular make command ? Which
directory are you running this from?
Should I perform the regular make command as: make j XX I still get the
same error. However, I do not understand why is this happening as I am into
the main application directory where the *Makefile* file exists. Unless, it
is the matter of generating another MOOSE application specifically for the
debugger only?

Are you using the conda installation, if so did you remember to activate
the conda environment?
For some reason internal to the business that I work for I am not sure that
the conda activation command is needed for the cluster environment but I
might be wrong on this and I would need to double check on that. In any
cases, by running the conda activate moose the output message is: *bash:
conda: command not found*

Have you used gdb before? The page you linked is the right one to help you.
As I said, I am relatively new to MOOSE and so far I have never used gdb
before. But to be honest, by reading the webpage about debugger it is
unclear to me about which MOOSE application to consider for the debugger.

I look forward to hearing from you.

Kind Regards,

Domenico



On Thu, 12 Aug 2021 at 21:35, Guillaume Giudicelli ***@***.***>
wrote:

> XX is the number of processes here.
>
> note the --args I added to the command btw
>
> Are you getting the same errors with the regular make command ?
> Which directory are you running this from? Are you using the conda
> installation, if so did you remember to activate the conda environment?
>
> Have you used gdb before? The page you linked is the right one to help you.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/18588#discussioncomment-1165528>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVFIBHYY4B6PN3HKXACAERTT4QWARANCNFSM5CA2NXMA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
*Domenico di Cugno* // Systems Engineer

*e: ***@***.***   *t*: 0117 906 1100   *w: *
www.cfms.org.uk
CFMS Services Ltd // Bristol & Bath Science Park // Dirac Crescent // Emersons
Green // Bristol // BS16 7FR

<https://cfms.org.uk/media/384108/email-footer-darren-swift.png?width=500&height=115&mode=crop>
<https://cfms.org.uk/media/384108/email-footer-darren-swift.png?width=500&height=115&mode=crop>
<https://cfms.org.uk/industry-themes/smart-infrastructure/?utm_source=email&utm_medium=gmail-footer&utm_campaign=infrastructure&utm_content=landing-page.>
<https://cfms.org.uk/industry-themes/smart-infrastructure/?utm_source=email&utm_medium=gmail-footer&utm_campaign=infrastructure&utm_content=landing-page.>
<https://cfms.org.uk/news-events-opinions/opinions/2021/may/make-the-most-of-automation-with-open-networking/?utm_source=email&utm_medium=gmail-footer&utm_campaign=opinions&utm_content=open-networking.>
<https://cfms.org.uk/news-events-opinions/opinions/2021/may/make-the-most-of-automation-with-open-networking/?utm_source=email&utm_medium=gmail-footer&utm_campaign=opinions&utm_content=open-networking.>
<https://cfms.org.uk/news-events-opinions/opinions/2021/may/make-the-most-of-automation-with-open-networking/?utm_source=email&utm_medium=gmail-footer&utm_campaign=opinions&utm_content=open-networking.>

**Reply by GiudGiud** [reply]
Hey 

Please use the quote reply feature (next to the checkmark button) if you want to quote. Will make it easier to read.

The XX for make and the XX for mpirun do not have to match. They will match if you want to use the same number of processes for the two separate tasks, and they tend to be both equal to the number of cores that your machine has, but they do not have to. 

Ok so the error does not have to do with generating the debug executable. Since you are not using conda, and are using a computing cluster where I assume someone else installed moose for you, what you need to do is specify where the MOOSE directory is before building your application. What does `echo $MOOSE_DIR` return?

Once this is fixed, we are trying to build a new executable specifically for debugging. Not really a new application. 

gdb is not tied to moose. It's used in general for debugging. You can find a lot of help on using it just by searching online. You should be good just using the command I gave you though, once you have built the application.

Guillaume

**Reply by DomenicoCFD** [reply]
Hi,

Many thanks for your clarifications. Here below are my thoughts:

> By running the echo $MOOSE_DIR I do receive only an empty line as output. I guess this is because we compile a specific version of moose and so, at every job and within the bash file, we execute a module load command as follows:

module load /path_to_module_files/moose-vxxx

So I guess this is what echo command should give.

> I can now see that it is not the case of generating a new application, but the question remain around the command I would need to generate the -gdb executable. Do I need to use the path to the moose somehow in any command line? as the METHOD= ... won't work.

Kind Regards,

Domenico

**Reply by GiudGiud** [reply]
Ugh the formatting of the message is messed up again. I guess I ll just have to deal with it ^^

Ok if you are to recompile the executable in debug mode you need to have access to moose. So module load, then compile.

When compiling, either MOOSE should be in the application/moose folder, as a submodule or MOOSE_DIR should be set.

**Reply by GiudGiud** [reply]
You can specify the path in the command line if you want to, so
MOOSE_DIR=path_to_moose METHOD=dbg make -j XX

**Reply by DomenicoCFD** [reply]
Hi,

It seems that something is now happening but still some errors as it cannot find few commands:
here below there is the output/error message.

[username@hostname application]$ MOOSE_DIR=/path_to_moose/moose-210516/ METHOD=dbg make -j 24
make: mpicxx: Command not found
make: mpif90: Command not found
/bin/sh: mpif90: command not found
Using HIT from /path_to_moose/framework/contrib/hit
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/src_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/outputs_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/outputs_png_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/outputs_formatters_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/controls_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/reporters_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/timeintegrators_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/fvbcs_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/meshgenerators_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/relationshipmanagers_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/vectorpostprocessors_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/problems_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/partitioner_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/timesteppers_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/auxkernels_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/fviks_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/predictors_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/constraints_Unity.C...
Compiling C++ (in dbg mode) /path_to_moose/framework/build/unity_src/transfers_Unity.C...
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/src_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
make: *** Waiting for unfinished jobs....
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/outputs_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/outputs_png_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/outputs_formatters_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/controls_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/reporters_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/timeintegrators_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/fvbcs_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/meshgenerators_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/relationshipmanagers_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/vectorpostprocessors_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/problems_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/partitioner_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/timesteppers_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/auxkernels_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/fviks_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/predictors_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/constraints_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
/path_to_moose/libmesh/installed/contrib/bin/libtool: line 1763: mpicxx: command not found
make: *** [/path_to_moose/framework/build/unity_src/transfers_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1

**Reply by GiudGiud** [reply]
It's not finding the mpi wrapped compilers, which will compile your application with the MPI library that you are using. 

You need to export the CC, CXX and FC environment variables to the location of your MPI-wrapped compilers.

If you don't know where they are, you'll need to talk to whoever installed MOOSE for you as you need to use the same compilers as those that were used to install MOOSE.

**Reply by DomenicoCFD** [reply]
Hi,

Many thanks for your direction! Really useful! I will speak tomorrow morning UTC time zone with whoever is/was involved into the MOOSE installation.

I will keep you informed on how it goes eventually! But for now many thanks for your priceless support!

Kind Regards,

Domenico
